x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -260,7 +260,7 @@ export class TaskManagerRunner implements TaskRunner {
         startedAt: now,
         attempts,
         retryAt: this.instance.schedule
-          ? intervalFromNow(this.definition.timeout)!
+          ? intervalFromNow(this.instance.schedule!.interval)!
```

Comment:
I'm curious what your thoughts are on my comment here: https://github.com/elastic/kibana/issues/83490#issuecomment-728346076.

> It seems like the `retryAt` should be the next scheduled run or the timeout window (whichever is greater) but that may break the timeout logic..

The only concern I had in regards to it is when an alert runs every 5 seconds or less and may get retried too early (in case it takes longer than interval to run).

Reply:
If someone is running a task that takes longer than X seconds, and is scheduled to run on an X second interval ... I dunno, all bets are off?

I'd guess we will eventually make the retry stuff more complicated anyway, eventually, (eg, add exponential backoff?) so we'll be in this again ... later, with even more interesting cases!

For now, the fix in this PR seems right to me.

Reply:
> If someone is running a task that takes longer than X seconds

My scenario would be the task running usually fast but taking longer when Elasticsearch is under pressure. I don't have strong preference either way but wanted to make sure we discuss such possibility.

Reply:
Yeah I went back and forth on this...

It wouldn't break the timeout logic as best as I can tell, but it has a potential pit fall - like Patrick noted, if a timeout is longer than the schedule, it will always trump the schedule. It also occurred to me that if we ever introduce a default timeout and _that_ is higher than a schedule specified by the user, then it will win.
I decided to leave it an open question for now rather than introduce a new behaviour that I'm unsure about.

That said, having read you scenario @mikecote I've actually gone 180 on this.
I think you're right- if a timeout _is specified_, then we should allow the task to run that long before retrying, so we should use the higher of both.  *But* it's worth noting - alerting tasks do not have a _timeout_, and we don't allow implementers to specify one, so alerting tasks will actually always short circuit on their schedule.

I will make the change for those tasks that _have_ a timeout.

Reply:
I'm thinking "no timeout = default timeout" in the scenarios above. So any task with a schedule would fall into option 4 otherwise fall into option 2.

The reason I'm thinking this path is because of "schedules" being a recommended next run (startAt + schedule) and if the task took longer to run than the schedule, it gets re-scheduled to run right away. Similar to how we don't do true cron like scheduling.

Note / question: [This](https://github.com/elastic/kibana/blob/master/x-pack/plugins/task_manager/server/task_running/task_runner.ts#L389) may want to do some sort of `Max.max(now, intervalFromDate(...))` otherwise it could re-schedule itself at the a higher level in the queue to run again if the task took longer than the interval to run.

> Perhaps we should also add support for specified `timeout` in alert definitions?

This is definitely something we should support at some point, if it's easy to add at the same time, I'm +1 on it otherwise an issue is perfectly fine.

Reply:
👍